base/src/main/java/com/fasterxml/jackson/jaxrs/cfg/MapperConfiguratorBase.java

```diff
@@ -81,27 +81,27 @@ public MapperConfiguratorBase(MAPPER mapper, Annotations[] defaultAnnotations)
     /***********************************************************
      */
 
-    public synchronized final void setMapper(MAPPER m) {
+    public final void setMapper(MAPPER m) {
         _mapper = m;
     }
 
-    public synchronized final void setAnnotationsToUse(Annotations[] annotationsToUse) {
+    public final void setAnnotationsToUse(Annotations[] annotationsToUse) {
```

Comment:
Just a quick note: these are not equivalent -- you are changing synchronization from applying to accessing mapper and configuring, to only covering mapper access and not config method.
Not sure it matters much in practice but just noting difference

Reply:
I think calling this method while other threads are using this instance is dangerous regardless of whether we have synchronization. Synchronization might sometimes help but its pot luck about the timing of the calls. This method should normally only be called when you are prepping the instance and not as an afterthought when you have started using it in other threads.

Reply:
Yes, agreed.